dali/util/uri.cc

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/util/uri.h"
+#include <stdexcept>
+
+namespace dali {
+
+ParsedURI ParseURI(const std::string& uri) {
+  auto pos = uri.find("://");
+  if (pos == std::string::npos)
+    throw std::runtime_error("Could not find a scheme in the URI: " + uri);
```

Comment:
A scheme is simply a bunch of letters followed by a colon - the `//` is not required in an URI.
`mailto:john@doe.com` is a valid URI - it just happens to not include the authority (the part starting with `//`).